ttt/lib/ttt/board.ex

```diff
@@ -12,30 +16,138 @@ defmodule TTT.Board do
     |> mark_with_fewest_moves
   end
 
-  def get_move_count_per_mark(board, marks) do
+  def game_over?(board) do
+    Enum.any?([not remaining_spaces?(board), found_winner?(board)], is_true?)
+  end
+
+  def winning_mark(board) do
+    winning_mark_in_row(board) <>
+     winning_mark_in_column(board) <>
+      winning_mark_in_diagonal(board)
+  end
+
+  def found_winner?(board) do
+    Enum.any?(win_in_any_direction?(board), is_true?)
+  end
+
+  defp win_in_any_direction?(board) do
+    [column_win?(board), row_win?(board), diagonal_win?(board)]
+  end
+
+  defp winning_mark_in_diagonal(board) do
+    diagonal_marks(board)
+    |> winning_mark_in_row
+  end
+
+  defp winning_mark_in_column(board) do
+    transpose(board)
+    |> winning_mark_in_row
+  end
+
+  defp winning_mark_in_row([]), do: ""
+  defp winning_mark_in_row([row | tail]) do
+    if not all_same_mark?([mark | _] = row) do
+      winning_mark_in_row(tail)
+    else
+      mark
+    end
+  end
+
+  def all_same_mark?([mark | t]) do
+    Enum.all?(t, fn(move) -> move == mark end)
+  end
+
+  def row_win?(board) do
+    found_winner_in_row?(board)
+  end
+
+  def column_win?(board) do
+    transpose(board)
+    |> row_win?
+  end
+
+  def diagonal_win?(board) do
+    diagonal_marks(board)
+    |> row_win?
+  end
+
+  defp found_winner_in_row?([], all_same), do: all_same
```

Comment:
`found_winner_in_row` feels complicated. Its not intuitive what the last boolean parameter is supposed to be. Why do I need more information than the row itself (the first parameter)?